### PR TITLE
Use protobuf and gRPC

### LINKED ISSUE #316: Use protobuf for consistent native API contracts

### LINKED ISSUE #321: Go to app switcher on iOS

### LINKED ISSUE #322: Toggle Wi-Fi on iOS

### LINKED ISSUE #357: Add a simple method to open the app under test
I've seen users open the app under test using `NativeAutomator.openApp()`:

```dart
void main() async {
  patrolTest(
    'main flow test',
    config: patrolConfig,
    nativeAutomation: true,
    ($) async {
      await $.pumpWidgetAndSettle(ExampleApp());
      
      // ...
      await $.native.openApp(id: patrolConfig.packageName!); // <--- not nice!
      // ...
    },
  );
}
```

We could improve this so it'd look like this:

```dart
void main() async {
  patrolTest(
    'main flow test',
    config: patrolConfig,
    nativeAutomation: true,
    ($) async {
      await $.pumpWidgetAndSettle(ExampleApp());
      
      // ...
      await $.native.openAppUnderTest(); // <--- nice!
      // ...
    },
  );
}
```

cc @iasiu: do you have better ideas for that new method?

### PR DESCRIPTION
**Pull request by bartekpacia**
- fix #316
- fix #322
- fix #357
- fix #321

### CODE REVIEW

File: AutomatorServer/android/app/src/androidTest/java/pl/leancode/automatorserver/PatrolAutomator.kt [hunk 1]
```diff
@@ -198,7 +187,7 @@ class PatrolAutomator {
         pressBack() // Hide keyboard.
     }
 
-    fun swipe(swipe: SwipeCommand) {
+    fun swipe(swipe: Contracts.SwipeCommand) {
```
Comment (bartekpacia) [inline AutomatorServer/android/app/src/androidTest/java/pl/leancode/automatorserver/PatrolAutomator.kt:190]: I hate this prefix but haven't found a way to get rid of this. Same on Swift side.

File: contracts.proto [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+message HandlePermissionCommand {
+  Code code = 1;
+
+  enum Code {
+    WHILE_USING = 0;
+    ONLY_THIS_TIME = 1;
+    DENIED = 2;
+  }
+}
```
Comment (bartekpacia) [inline contracts.proto:50]: all the "commands" and "queries" are gonna get GRPC-ed in the next PR (or maybe this one? idk).

**Review by Xezolpl — CHANGES_REQUESTED** [review]

File: AutomatorServer/ios/AutomatorServerUITests/PatrolAutomation.swift [hunk 3]
```diff
@@ -99,28 +105,62 @@ class PatrolAutomation {
...
+      }
+      
+      self.springboard.activate()
+      self.preferences.terminate()
+      XCUIApplication(bundleIdentifier: bundleIdentifier).activate() // go back to the app under test
+    }
+  }
+  
+  func disableWiFi(_ bundleIdentifier: String) {
+    runAction("enabling wifi") {
```
Comment (Xezolpl) [inline AutomatorServer/ios/AutomatorServerUITests/PatrolAutomation.swift:127]:
```suggestion
    runAction("disabling wifi") {
```

**Review by Xezolpl — CHANGES_REQUESTED** [review]

File: gen_protobufs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+kotlin_out="./AutomatorServer/android/app/src/androidTest/java"
+swift_out="./AutomatorServer/ios/AutomatorServerUITests"
+
+protoc --dart_out="grpc:$dart_out" -I="$SRC_DIR" "$PROTO_PATH"
+
+protoc \
+	--java_out="$java_out" \
+	--kotlin_out="$kotlin_out" \
+	--grpc-java_out="$java_out" \
+    --grpckt_out="$kotlin_out" \
```
Comment (jakubfijalkowski) [inline gen_protobufs:34]: Indent

File: AutomatorServer/ios/AutomatorServerUITests/PatrolAutomation.swift [hunk 5]
```diff
@@ -22,25 +22,31 @@ class PatrolAutomation {
...
   }
-
+  
   func pressHome() {
     runAction("pressing home button") {
       self.device.press(XCUIDevice.Button.home)
     }
   }
-
+
```
[the remark this answers is not in the document]
Comment (bartekpacia) [reply AutomatorServer/ios/AutomatorServerUITests/PatrolAutomation.swift:34]: yeah, Xcode moment.

I used to treat everything with `swift-format` but after upgrading to Xcode 14, `swift-format` kind of died and refuses to work.

File: AutomatorServer/ios/AutomatorServerUITests/PatrolAutomation.swift (hunk 3 above)
Comment (bartekpacia) [reply AutomatorServer/ios/AutomatorServerUITests/PatrolAutomation.swift:127]: Haha, thanks!

File: gen_protobufs (hunk 4 above)
Comment (bartekpacia) [reply gen_protobufs]: fix'd

File: packages/patrol/example/integration_test/ios/devel_test.dart [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+      await $.pumpWidgetAndSettle(ExampleApp());
+
+      await $.native.pressHome();
+      await $.native.pressRecentApps();
+
+      await $.native.openApp(appId: 'pl.leancode.patrol.Example');
+
+      await _wait();
+
+      $.log('test end');
```
[the remark this answers is not in the document]
Comment (bartekpacia) [reply packages/patrol/example/integration_test/ios/devel_test.dart]: nowhere:P

this file was a copy-pasted debugging aid. I removed it now. Thanks.

**Pull request merged**